### ISSUE TITLE
[BUG] dangling-servicemonitor ignores the monitor namespace by default

### ISSUE THREAD

**Issue by lokesh0186** [stackrox/kube-linter#1256]
**System info:**

- OS: macOS 26.5.1, arm64
- kube-linter: v0.8.3
- Also confirmed in current `main` source at `ea3abb8abfe0fc4d7c2b20c9a30d95db9f0b2e2a`

**Describe the bug**

The `dangling-servicemonitor` check accepts a Service from a different namespace when the ServiceMonitor omits `spec.namespaceSelector`.

Prometheus Operator defines an empty `NamespaceSelector` as selecting objects from the current namespace. In this example, the ServiceMonitor is in `monitoring`, so the Service in `application` should not satisfy it.

**To Reproduce**

Save the YAML below as `repro.yaml`, then run:

```console
kube-linter lint repro.yaml --do-not-auto-add-defaults --include dangling-servicemonitor
```

The v0.8.3 binary exits 0 with:

```text
KubeLinter 0.8.3

No lint errors found!
```

**Sample YAML input**

```yaml
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: application-metrics
  namespace: monitoring
spec:
  selector:
    matchLabels:
      app: application
  endpoints:
    - port: metrics
---
apiVersion: v1
kind: Service
metadata:
  name: application
  namespace: application
  labels:
    app: application
spec:
  selector:
    app: application
  ports:
    - name: metrics
      port: 9090
      targetPort: metrics
```

**Expected behavior**

The check should report `application-metrics` as dangling because its empty namespace selector restricts discovery to `monitoring`, where no matching Service exists.

Two controls behave as expected:

- Moving the Service to `monitoring` produces no finding.
- Adding `namespaceSelector.matchNames: [application]` to the ServiceMonitor produces no finding.

**Additional context**

The current implementation marks the namespace selector as unset, then [returns success for any label-matching Service](https://github.com/stackrox/kube-linter/blob/ea3abb8abfe0fc4d7c2b20c9a30d95db9f0b2e2a/pkg/templates/danglingservicemonitor/template.go#L35-L68), regardless of that Service's namespace. The Prometheus Operator API states that empty `matchNames` with `any: false` [selects from the current namespace](https://github.com/prometheus-operator/prometheus-operator/blob/ff5cad994c539b3568b314f3c5086275e93800f4/pkg/apis/monitoring/v1/types.go#L792-L805).

A focused fix would preserve the existing `any` and `matchNames` behavior while treating an empty namespace selector as the ServiceMonitor's namespace. A regression test with the explicit monitor and Service namespaces above should cover the false negative.

As an independent relationship cross-check, public [`iac-guard-v==0.1.0b1`](https://pypi.org/project/iac-guard-v/0.1.0b1/) reports `VIOLATED` with `SERVICEMONITOR_TARGET_UNRESOLVED` for this case, and `SATISFIED` for both controls.
